libs/core/src/lib/segmented-button/segmented-button.component.ts

```diff
@@ -90,8 +93,15 @@ export class SegmentedButtonComponent implements AfterViewInit, ControlValueAcce
     constructor(
         private readonly _changeDetRef: ChangeDetectorRef,
         private readonly _elementRef: ElementRef,
-        private readonly _destroyRef: DestroyRef
-    ) {}
+        private readonly _destroyRef: DestroyRef,
+        @Optional() @Host() private _focusableList: FocusableListDirective,
+        @Optional() private _rtlService: RtlService
+    ) {
+        this._focusableList.navigationDirection = this.vertical ? 'vertical' : 'horizontal';
```

Comment:
This will need to be updated reactively, what if in the runtime user replaced vertical with horizontal?